packages/gluestick/shared/lib/getHttpClient.js

```diff
@@ -47,7 +47,7 @@ export default function getHttpClient(
   // If a request object is provided, then we want to merge the custom headers
   // with the headers that we sent from the browser in the request.
   client = httpClient.create({
-    baseURL: protocol + req.headers.host,
+    baseURL: protocol + req.hostname,
```

Comment:
This will fallback to `req.get('host')` when `trust proxy` is not enabled: 

https://github.com/expressjs/express/blob/4.16.4/lib/request.js#L431-L433